lib/libc/minimal/source/stdlib/malloc.c

```diff
@@ -35,7 +35,12 @@ Z_GENERIC_SECTION(POOL_SECTION) static char z_malloc_heap_mem[HEAP_BYTES];
 void *malloc(size_t size)
 {
 	void *ret;
-	ret = sys_heap_alloc(&z_malloc_heap, size);
+	/*
+	 * By aligning to Z_MAX_ALIGN, we guarantee, in a portable way, that
+	 * the returned pointer can be used to access any of the types in
+	 * z_max_align_t.
+	 */
+	ret = sys_heap_aligned_alloc(&z_malloc_heap, Z_MAX_ALIGN, size);
```

Comment:
As mentioned, this criteria is already met.  I'd really prefer this be a build-time check vs. increasing code size.